portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,18 +98,68 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty("commentInput", commentInput);
     commentEntity.setProperty("timestampMillis", timestampMillis);
 
+    // Handle file upload input from form.
+    getUploadedFileUrl(request, "file").ifPresent(fileUrl -> commentEntity.setProperty("fileUrl", fileUrl));
+    
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity); 
 
     // Redirect back to the blog page.
     response.sendRedirect("/blog.html");
   }
 
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+  private static String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
     if (value == null) {
       return defaultValue;
     }
     return value;
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private static Optional<String> getUploadedFileUrl(HttpServletRequest request, 
+                                                     String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return Optional.empty();
+    }
+
+    // Make sure there is only one input.
+    if (blobKeys.size() != 1) {
+      throw new IllegalStateException("There should only be one file input.");
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return Optional.empty();
+    }
+
+    // We could check the validity of the file here, e.g. to make sure it's an image file
+    // https://stackoverflow.com/q/10779564/873165
```

Comment:
For things like this add it as a TODO if it's somoething you want to do later.

But if it's explaining what you're doing below remove the. "could check" since it makes it seem like you havent done it yet.